### PR TITLE
feat: image zoom dialog

### LINKED ISSUE #7919: Create zoomed modal
- Create the modal that will contain the zoom image image and thumbnails
- Add the trigger button in the product image
- Add a configuration to mitigate breaking changes

### Acceptance

- Clicking the trigger opens an empty dialog
- Dialog is configurable using the launch mechanism

### PR DESCRIPTION
**Pull request by LTiger14**
closes #7919

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/product/product-images/product-images.module.ts [hunk 1]
```diff
@@ -14,6 +16,8 @@ import { ProductImagesComponent } from './product-images.component';
     MediaModule,
     OutletModule,
     CarouselModule,
+    IconModule,
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-images/product-images.module.ts:19]: `ImageZoomModule` imports `IconModule`. So, here no need import `IconModule`.

File: projects/storefrontlib/src/shared/components/image-zoom/dialog/image-zoom-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { ICON_TYPE } from '../../../../cms-components/misc/icon/icon.model';
+import { FocusConfig } from '../../../../layout/a11y/index';
+import { LaunchDialogService } from '../../../../layout/launch-dialog/services/index';
+
+@Component({
+  selector: 'cx-image-zoom-dialog',
+  templateUrl: 'image-zoom-dialog.component.html',
+})
+export class ImageZoomDialogComponent {
+  iconTypes = ICON_TYPE;
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/shared/components/image-zoom/dialog/image-zoom-dialog.component.ts:11]: `iconType` is better than `iconTypes`.

File: projects/storefrontlib/src/shared/components/image-zoom/trigger/image-zoom-trigger.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  expandImage(): void {
+    const component = this.launchDialogService.launch(
+      LAUNCH_CALLER.IMAGE_ZOOM,
+      this.vcr
+    );
+    if (component) {
+      this.subscriptions.add(
+        combineLatest([component, this.launchDialogService.dialogClose])
+          .pipe(
+            filter(([, close]) => close && close !== undefined),
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/shared/components/image-zoom/trigger/image-zoom-trigger.ts:30]: for this one: `close && close !== undefined`. if `close` is true, no need check this `close !== undefined`, right?

**Review by WeizhengSap — APPROVED** [review]

File: projects/storefrontlib/src/shared/components/image-zoom/trigger/image-zoom-trigger.ts (hunk 3 above)
Comment (LTiger14) [reply projects/storefrontlib/src/shared/components/image-zoom/trigger/image-zoom-trigger.ts:30]: Yes it should

**Pull request merged**